cpp/src/copying/slice.cpp

```diff
@@ -36,7 +36,22 @@ std::vector<column_view> slice(column_view const& input,
 
   if (indices.empty()) return {};
 
-  auto null_counts = cudf::detail::segmented_count_unset_bits(input.null_mask(), indices, stream);
+  auto null_counts = [&]() {
+    // need to shift incoming indices by the column offset to generate the correct bit ranges
+    // to count
+    if (input.offset() > 0) {
+      std::vector<size_type> shifted_indices;
+      shifted_indices.reserve(indices.size());
+      std::transform(indices.begin(),
+                     indices.end(),
+                     std::back_inserter(shifted_indices),
+                     [offset = input.offset()](size_type index) { return index + offset; });
+      return cudf::detail::segmented_count_unset_bits(input.null_mask(), shifted_indices, stream);
+    }
+    // can use the initial indices
+    return cudf::detail::segmented_count_unset_bits(input.null_mask(), indices, stream);
+  }();
+
```

Comment:
@nvdbaranec I see that the function signature `std::vector<size_type> segmented_count_unset_bits(bitmask_type const* bitmask,....)` accepting a pointer as its first argument. Can we just use this instead?
```
auto const null_counts = cudf::detail::segmented_count_unset_bits(input.null_mask() + input.offset(), indices, stream);
```

Reply:
No.  The null_mask is of type bitmask_type (32 bits).  32 rows per bitmask_type word, so you can't just add offset.